lib/cocoapods/installer/user_project_integrator/target_integrator.rb

```diff
@@ -130,6 +131,94 @@ def add_pods_library
           end
         end
 
+        # Add a reference to pods' resources (like xcassets) to the user target
+        #
+        # @return [Boolean] true if user project has been modified
+        #
+        def add_pods_resources
+          dirty = false
+        
+          UI.puts "- User targets being integrated: #{native_targets.map(&:name).inspect}"
+          UI.puts "- Pod Targets = #{target.pod_targets.map(&:name)}"
+
+          pods_group = user_project['Pods']
+          resources_group = pods_group ? pods_group['Resources'] : nil
+           # The files already in the target before integration
+          refs_to_remove = []
+          if resources_group
+            # Compute the list of file_refs form the Resources/ group that are 
+            # also in the native_targets Resources Build Phase
+            copied_resources_files = native_targets.map(&:resources_build_phase).flat_map(&:files_references)
+            copied_resources_paths = copied_resources_files.map(&:path).uniq
+            resources_files = resources_group.groups.flat_map(&:files)
+            refs_to_remove = resources_files.select { |ref| copied_resources_paths.include?(ref.path) }
+          end
+
+          target.pod_targets.each do |pod_target|
+            pod_name = pod_target.pod_name
+            resource_files = pod_target.file_accessors.flat_map(&:resources)
+            resource_files.each do |resource_path|
+              UI.puts "   - Adding #{resource_path} to user project"
+
+              relative_path = resource_path.relative_path_from(target.client_root).to_s
+              pods_group ||= ((dirty = true) && user_project.new_group('Pods'))
+              resources_group ||= ((dirty = true) && pods_group.new_group('Resources'))
+              pod_subgroup = resources_group[pod_name] || ((dirty = true) && resources_group.new_group(pod_name))
+              file_ref = pod_subgroup.files.find { |f| f.path == relative_path }
+              file_ref ||= (dirty = true) && pod_subgroup.new_file(relative_path)
+
+              UI.puts "   - Adding #{resource_path.basename} to targets #{native_targets.map(&:name)}"
+              native_targets.each do |user_target|
+                refs_to_remove.delete(file_ref) # this file_ref is still needed, don't remove it later
+                unless user_target.resources_build_phase.include?(file_ref)
+                  user_target.add_resources([file_ref])
+                  dirty = true
+                end
+              end
+            end
+          end
+
+          # Remove the resources no longer in a target
+          refs_to_remove.each do |file_ref|
+            native_targets.each do |user_target|
+              UI.puts "   - Removing #{file_ref} from #{user_target} because it is no longer needed."
+              # FIXME: XcodeProj issue? The PBXBuildFile is replaced with (null) instead of being removed
```

Comment:
Anyone having an idea where this XcodeProj bug comes from?

(Probably needs some more tests to isolate in a simpler sample project / test case)